stages/org.osbuild.ostree.deploy

```diff
@@ -107,30 +124,68 @@ def main(tree, options):
     kopts = options.get("kernel_opts", [])
     ref = options["ref"]
     remote = options.get("remote")
-
-    if remote:
-        ref = f"{remote}:{ref}"
+    is_container = options.get("container")
 
     kargs = []
 
-    if rootfs:
-        rootfs_id = make_fs_identifier(rootfs)
-        kargs += [f"--karg=root={rootfs_id}"]
+    if (not is_container):
+      if remote:
+          ref = f"{remote}:{ref}"
 
-    for opt in kopts:
-        kargs += [f"--karg-append={opt}"]
+      if rootfs:
+          rootfs_id = make_fs_identifier(rootfs)
+          kargs += [f"--karg=root={rootfs_id}"]
+
+      for opt in kopts:
+          kargs += [f"--karg-append={opt}"]
+
+    else:
+      for opt in kopts:
+          kargs += [f"--karg={opt}"]
 
     with MountGuard() as mounter:
         for mount in mounts:
             path = mount.lstrip("/")
             path = os.path.join(tree, path)
             mounter.mount(path, path)
 
-        ostree("admin", "deploy", ref,
-               *kargs,
-               sysroot=tree,
-               os=osname)
-
+        if (not is_container):
+          ostree("admin", "deploy", ref,
+                 *kargs,
+                 sysroot=tree,
+                 os=osname)
+        
+        else:
+          # create a temporary directory to store the ostree commit id after deploying the container image
+          os.mkdir(f"{tree}/tmp")
+          
+          # parse and construct the imgref of the container image pulled during the skopeo stage
+          destination = options["container-source"]
+          storage_root = destination.get("storage-path", "/var/lib/containers/storage")
+          storage_driver = destination.get("storage-driver", "overlay")
+
+          imgref = f"{remote}:containers-storage:[{storage_driver}@{tree}{storage_root}+/run/containers/storage]{ref}"
+          
+          extra_args = []
+          if remote:
+              extra_args.append(f'--imgref={imgref}')
+              extra_args.append(f'--stateroot={osname}')
+              extra_args.append(f'--write-commitid-to={tree}/tmp/commit.txt')
+
+          ostree("container", "image", "deploy",
+                *extra_args,
+                sysroot=tree, 
+                *kargs)
+          
+          f = open(f'{tree}/tmp/commit.txt', 'r')
+          commit = f.read()
+          target = f"{tree}/ostree/deploy/{osname}/deploy/{commit}.0.origin"
+          
+          end_of_line = '$d'
+          subprocess.run(["sed", "-i", f"{end_of_line}", target], check=True)
+          
+          remote_source = '$acontainer-image-reference=ostree-unverified-registry:quay.io/fedora/fedora-coreos:stable'
+          subprocess.run(["sed", "-i", f"{remote_source}", target], check=True)
```

Comment:
I spotted the code that makes this `sed` not needed. See the `--target-imgref` in https://github.com/coreos/coreos-assembler/blob/cd3937f80f3b39ae620eb3f15b84ad76809e7d47/src/create_disk.sh#L342